gateway/src/shard/tls.rs

```diff
@@ -0,0 +1,161 @@
+//! TLS manager to reuse connections between shards.
+
+#[cfg(feature = "rustls")]
+use std::sync::Arc;
```

Comment:
this directive will only affect the Arc import, we should put it on a separate line to make it clear